### PR TITLE
Update galleries to use Multi-weight idioms

### PR DESCRIPTION
**Pull request by datumbox**
Resolves partially #6014

This PR:
- Updates the preprocessing description for RAFT
- Fixes the incorrect examples for detection models (passing tensors vs List[tensor])
- Updates the preprocessing descriptions and the meta-data usage of viz-utils gallery

### CODE REVIEW

File: gallery/plot_optical_flow.py [hunk 1]
```diff
@@ -81,11 +81,11 @@ def plot(imgs, **imshow_kwargs):
...
-# The RAFT model that we will use accepts RGB float images with pixel values in
-# [-1, 1]. The frames we got from :func:`~torchvision.io.read_video` are int
-# images with values in [0, 255], so we will have to pre-process them. We also
-# reduce the image sizes for the example to run faster. Image dimension must be
-# divisible by 8.
+# The RAFT model accepts RGB images. We first get the frames from
+# :func:`~torchvision.io.read_video` and resize them to ensure their
+# dimensions are divisible by 8. Then we use the transforms bundled into the
+# weights in order to preprocess the input and rescale its values to the
+# required ``[-1, 1]`` interval.
```
Comment (datumbox) [inline gallery/plot_optical_flow.py:88]: I rewrote this to reflect what preprocessing is currently visible to the user while using the presets.

File: gallery/plot_visualization_utils.py [hunk 2]
```diff
@@ -43,8 +43,9 @@ def show(imgs):
 
 dog1_int = read_image(str(Path('assets') / 'dog1.jpg'))
 dog2_int = read_image(str(Path('assets') / 'dog2.jpg'))
+dog_list = [dog1_int, dog2_int]
 
-grid = make_grid([dog1_int, dog2_int, dog1_int, dog2_int])
+grid = make_grid(dog_list)
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:48]: Not sure why we were showing twice the picture of each dog previously.

File: gallery/plot_visualization_utils.py [hunk 3]
```diff
@@ -65,28 +66,23 @@ def show(imgs):
 
 #####################################
 # Naturally, we can also plot bounding boxes produced by torchvision detection
-# models.  Here is demo with a Faster R-CNN model loaded from
+# models.  Here is a demo with a Faster R-CNN model loaded from
 # :func:`~torchvision.models.detection.fasterrcnn_resnet50_fpn`
-# model. You can also try using a RetinaNet with
-# :func:`~torchvision.models.detection.retinanet_resnet50_fpn`, an SSDlite with
-# :func:`~torchvision.models.detection.ssdlite320_mobilenet_v3_large` or an SSD with
-# :func:`~torchvision.models.detection.ssd300_vgg16`. For more details
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:73]: I'm not sure why we kept adding more and more models here but I ended up removing them all. I think people get it that they can use other models too, without listing them all.

File: gallery/plot_visualization_utils.py [hunk 4]
```diff
@@ -65,28 +66,23 @@ def show(imgs):
...
-# on the output of such models, you may refer to :ref:`instance_seg_output`.
+# model. For more details on the output of such models, you may
+# refer to :ref:`instance_seg_output`.
 
 from torchvision.models.detection import fasterrcnn_resnet50_fpn, FasterRCNN_ResNet50_FPN_Weights
 
 
-batch_int = torch.stack([dog1_int, dog2_int])
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:79]: This was the wrong way to use detection models. They receive a list of tensors rather than a batch, so I made the change everywhere below. I've also made sure not to reference `batch_int` in other places.

File: gallery/plot_visualization_utils.py [hunk 5]
```diff
@@ -131,8 +121,8 @@ def show(imgs):
 model = fcn_resnet50(weights=weights, progress=False)
 model = model.eval()
 
-normalized_batch = transforms(batch)
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:134]: Previously we were doubly preprocessing the images. I've changed it to do it only once using the original data (dog_list).

File: gallery/plot_visualization_utils.py [hunk 6]
```diff
@@ -145,18 +135,13 @@ def show(imgs):
 # Let's plot the masks that have been detected for the dog class and for the
 # boat class:
 
-sem_classes = [
-    '__background__', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus',
-    'car', 'cat', 'chair', 'cow', 'diningtable', 'dog', 'horse', 'motorbike',
-    'person', 'pottedplant', 'sheep', 'sofa', 'train', 'tvmonitor'
-]
-sem_class_to_idx = {cls: idx for (idx, cls) in enumerate(sem_classes)}
+sem_class_to_idx = {cls: idx for (idx, cls) in enumerate(weights.meta["categories"])}
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:138]: Switch to meta-data usage.

File: gallery/plot_visualization_utils.py [hunk 7]
```diff
@@ -252,7 +237,7 @@ def show(imgs):
 # Instance segmentation models
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
-# Instance segmentation models have a significantly different output from the
+# Instance segmentation models have a slightly different output from the
```
Comment (datumbox) [inline gallery/plot_visualization_utils.py:240]: I'm not sure why we said that MaskRCNN has a significantly different output than FasterRCNN here. It just includes the same as FasterRCNN + the mask info.

**Review by datumbox — COMMENTED** [review]
Few comments to highlight important changes:

File: gallery/plot_visualization_utils.py (hunk 3 above)
Comment (NicolasHug) [reply gallery/plot_visualization_utils.py:73]: It was for the backlinks. See https://pytorch.org/vision/main/models/generated/torchvision.models.detection.retinanet_resnet50_fpn.html#torchvision.models.detection.retinanet_resnet50_fpn at the bottom, theres a "Examples using retinanet_resnet50_fpn:" section.

Not mentioning the model builders in the example removes that. I don't have a strong opinion, I agree it's not really scalable if we have to remember to link all models from the examples. But it's also nice to have the backlinks... IDK. Maybe we can do that more manually.

File: gallery/plot_visualization_utils.py (hunk 2 above)
Comment (NicolasHug) [reply gallery/plot_visualization_utils.py:48]: I'd be in favour of removing this section about `make_grid` entirely. We shouldn't encourage our users to use it anymore, there are much better alternatives

File: gallery/plot_visualization_utils.py (hunk 7 above)
Comment (NicolasHug) [reply gallery/plot_visualization_utils.py:240]: The masks are quite different: one is a non-normalized score while for the other it's probabilities directly. Also one can detect an arbiratry number of classes while for the other one it's fixed, right?

(I'm fine with the change regardless)

File: gallery/plot_visualization_utils.py [hunk 8]
```diff
@@ -304,30 +289,15 @@ def show(imgs):
...
-    'bottle', 'N/A', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl',
-    'banana', 'apple', 'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza',
-    'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed', 'N/A', 'dining table',
-    'N/A', 'N/A', 'toilet', 'N/A', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
-    'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'N/A', 'book',
-    'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush'
-]
-
-inst_class_to_idx = {cls: idx for (idx, cls) in enumerate(inst_classes)}
+inst_class_to_idx = {cls: idx for (idx, cls) in enumerate(weights.meta["categories"])}
```
Comment (NicolasHug) [inline gallery/plot_visualization_utils.py:297]: should we print it?

File: gallery/plot_visualization_utils.py (hunk 3 above)
Comment (datumbox) [reply gallery/plot_visualization_utils.py:73]: Let's add manually, maybe with refs from the documentation pages of the methods as you recommended previously offline. The specific list was not complete and missing several new models.

File: gallery/plot_visualization_utils.py (hunk 7 above)
Comment (datumbox) [reply gallery/plot_visualization_utils.py:240]: happy to revert

File: gallery/plot_visualization_utils.py (hunk 2 above)
Comment (datumbox) [reply gallery/plot_visualization_utils.py:48]: Sounds good to me, let's do this on a separate PR to explore what the better recommended alternative is. cc @oke-aditya

File: gallery/plot_visualization_utils.py (hunk 8 above)
Comment (datumbox) [reply gallery/plot_visualization_utils.py:297]: Good spot. Actually I don't know why this was here. We never printed it. I think I'll just remove it rather than printing a big mapping table on the docs.
